From 7d629cfd045eab6fd70e3df4fbaa4881ac9e8956 Mon Sep 17 00:00:00 2001
From: Manish Dharanenthiran <quic_mdharane@quicinc.com>
Date: Tue, 16 Jul 2024 22:35:33 +0530
Subject: [PATCH] QSDK: hostapd: Fix NL80211 attribute mismatch

While adding support for CW and WLAN interference, new NL attribute is
introduced in the kernel. But, that attribute is not updated to hostapd
copy of NL attributes which leads to mismatch for any new addition of
NL80211 attributes.

Fix this by syncing NL80211 properly with kernel changes.

Fixes: I6acee3318 (QSDK: wifi: Add support for CW and WLAN interference)
Depends-on: I6acee3318 (QSDK: wifi: Add support for CW and WLAN interference)

Signed-off-by: Manish Dharanenthiran <quic_mdharane@quicinc.com>
---
 src/drivers/nl80211_copy.h | 1 +
 1 file changed, 1 insertion(+)

--- a/src/drivers/nl80211_copy.h
+++ b/src/drivers/nl80211_copy.h
@@ -1586,6 +1586,11 @@ enum nl80211_commands {
 	NL80211_CMD_AWGN_DETECT,
 
 	NL80211_CMD_STOP_BGRADAR_DETECT,
+
+	/* To be Deprecated from ATH QSDK, once we upstream
+	 * NL80211_CMD_INTERFERENCE_DETECT
+	 */
+	NL80211_CMD_INTERFERENCE_DETECT,
 	/* add new commands above here */
 
 	/* used to define NL80211_CMD_MAX below */
@@ -3456,6 +3461,7 @@ enum nl80211_attrs {
 	NL80211_ATTR_SET_CRITICAL_UPDATE,
 	NL80211_ATTR_CHANNEL_WIDTH_DEVICE,
 	NL80211_ATTR_CENTER_FREQ_DEVICE,
+	NL80211_ATTR_INTERFERENCE_TYPE,
 
 /* add attributes here, update the policy in nl80211.c */
 
